lib/backend/migration/migration.go

```diff
@@ -0,0 +1,155 @@
+package migration
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/utils/retryutils"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/trace"
+)
+
+// Migration manages a migration between two [backend.Backend] interfaces.
+type Migration struct {
+	src      backend.Backend
+	dst      backend.Backend
+	parallel int
+	total    int
+	migrated atomic.Int64
+	log      logrus.FieldLogger
+}
```

Comment:
What's the point of this object, other than to forget to close the backends at the end?

Reply:
What do you think something like this?

```
func Migrate(context.Context, MigrateConfig) error
```